src/rantanplan/rhymes.py

```diff
@@ -255,10 +260,14 @@ def analyze_rhyme(lines, offset=4):
                 stressed_endings, assonance, relaxation, offset
             )
             rhyme = "".join(rhymes)
-            rhythmical_lengths = [line["rhythm"]["length"] for line in lines]
-            ranking = search_structure(rhyme, rhythmical_lengths, rhyme_type)
-            if ranking is not None and ranking < best_ranking:
-                best_ranking = ranking
+            # rhythmical_lengths = [line["rhythm"]["length"] for line in lines]
+            length_ranges = [
+                range(line["rhythm"]["length_range"]["min_length"],
+                      line["rhythm"]["length_range"]["max_length"] + 1) for line
+                in lines]
+            ranking = search_structure(rhyme, length_ranges, rhyme_type)
+            if ranking is not None and ranking[0] < best_ranking:
+                best_ranking = ranking[0]
```

Comment:
Good. I understand that at some point not only the first match will be handled.
```suggestion
            ranking, _* = search_structure(rhyme, length_ranges, rhyme_type)
            if ranking is not None and ranking < best_ranking:
                best_ranking = ranking
```

Reply:
Still thinking on how to handle other candidates...

Reply:
If ranking is None (and it can be) that will fail. What about this:
```
            candidates = search_structure(rhyme, length_ranges, rhyme_type)
            if candidates is not None:
                ranking, *_ = candidates
            if ranking is not None and ranking < best_ranking:
                best_ranking = ranking
```
That way we save all candidates for later

Reply:
Although I like the returning of `[None]`, the first options seems clearer. Moreover, if should never return `None` but an empty list now.

Reply:
Returning an empty list would cause the _ assignment to fail

Reply:
Also, done with the first method, but the  search_structure return has to look like this: `return indices if indices else None` because of the reason I mentioned on my last reply

Reply:
I don't understand. Before these changes, the function returned a match or `None` if not found. If now instead of a match it returns a list, if there is no match it should return an empty list. If there is no match both a `None` and a `[]` will break the `*_`, that's because you added a check first to see for `candidates`.

Reply:
Ok, I'll revert some changes to reflect that, I think I made a mess...